### PR TITLE
add vestingPerBlock and vestingEndBlock in vested balance tooltip

### PR DESCRIPTION
**Pull request by niklabh**
https://github.com/polkadot-js/apps/issues/3561

Screenshot from 2020-09-01 21-28-11

### CODE REVIEW

**Comment by jacogr** [comment]
The end-block converted to time would be really nice.

**Comment by niklabh** [comment]
> The end-block converted to time would be really nice.

In which section should this be added. @Swader asking to be added to locked section. But isn't it a vesting info.

**Comment by Swader** [comment]
I think it kinda is, it signifies when the locked status will disappear

**Comment by niklabh** [comment]
> The end-block converted to time would be really nice.

For showing time we need remaining blocks count. Should have calculated and passed vestingRemainingBlocks in https://github.com/polkadot-js/api/pull/2563. Should i make change in api.

**Comment by jacogr** [comment]
endBlock - bestBlock = remaining.

File: packages/react-components/src/AddressInfo.tsx [hunk 1]
```diff
@@ -274,7 +274,7 @@ function renderBalances (props: Props, allAccounts: string[], t: <T = string> (k
                   <div className='faded'>{t('available to be unlocked')}</div>
                   {formatBalance(balancesAll.vestingPerBlock)}
                   <div className='faded'>{t('per block')}</div>
-                  <div className='faded'>{`${t('until block')} ${balancesAll.vestingEndBlock}`}</div>
+                  <div className='faded'>{`${t('until block')} ${balancesAll.vestingEndBlock.toString()}`}</div>
```
Comment (jacogr) [inline packages/react-components/src/AddressInfo.tsx:277]: Was hoping that we could have an end date like the calendar (or at least countdown via `BlockToTime`), however if we do display just the blockNumber, display it via `formatNumber(balancesAll.vestingEndBlock)` to align with the rest of the UI formatting.

Comment (niklabh) [reply packages/react-components/src/AddressInfo.tsx:277]: yes adding. This needed to be merged to show remaining blocks BlockToTime https://github.com/polkadot-js/api/pull/2566/files

Comment (niklabh) [reply packages/react-components/src/AddressInfo.tsx:277]: Then this needs to be added in this line. 

`<div className='faded'>{`${t('until block')} ${formatNumber(balancesAll.vestingEndBlock)} ${t('on')} `} <BlockToTime blocks={balancesAll.vestingRemainingBlocks} /></div>`

Don't want to push this until https://github.com/polkadot-js/api/pull/2566/files is merged so to not break the PR

Comment (niklabh) [reply packages/react-components/src/AddressInfo.tsx]: done

File: packages/react-components/src/AddressInfo.tsx [hunk 2]
```diff
@@ -3,16 +3,16 @@
 // of the Apache-2.0 license. See the LICENSE file for details.
 
 import { DeriveBalancesAll, DeriveDemocracyLock, DeriveStakingAccount } from '@polkadot/api-derive/types';
-import { LockIdentifier, ValidatorPrefsTo145 } from '@polkadot/types/interfaces';
+import { LockIdentifier, ValidatorPrefsTo145, BlockNumber } from '@polkadot/types/interfaces';
```
Comment (jacogr) [inline packages/react-components/src/AddressInfo.tsx:6]:
```suggestion
import { BlockNumber, LockIdentifier, ValidatorPrefsTo145 } from '@polkadot/types/interfaces';
```

**Review by jacogr — APPROVED** [review]
Thank you.

File: packages/react-components/src/AddressInfo.tsx [hunk 3]
```diff
@@ -269,7 +269,18 @@ function renderBalances (props: Props, allAccounts: string[], t: <T = string> (k
...
+                <div>
+                  {formatBalance(balancesAll.vestedClaimable, { forceUnit: '-' })}
+                  <div className='faded'>{t('available to be unlocked')}</div>
+                  {formatBalance(balancesAll.vestingPerBlock)}
+                  <div className='faded'>{t('per block')}</div>
+                  <div className='faded'>
+                    {`${t('until block')} ${formatNumber(balancesAll.vestingEndBlock)} `}
+                    <BlockToTime
+                      blocks={balancesAll.vestingEndBlock.sub(bestNumber)}
+                      isInline />
```
Comment (jacogr) [inline packages/react-components/src/AddressInfo.tsx:281]:
```suggestion
                    <div>{`${t('until block')} ${formatNumber(balancesAll.vestingEndBlock)}`}<div>
                    <BlockToTime blocks={balancesAll.vestingEndBlock.sub(bestNumber)} />
```

File: packages/react-components/src/AddressInfo.tsx [hunk 4]
```diff
@@ -269,7 +269,16 @@ function renderBalances (props: Props, allAccounts: string[], t: <T = string> (k
...
             <Tooltip
               text={
-                <div>{formatBalance(balancesAll.vestedClaimable, { forceUnit: '-' })}<div className='faded'>{t('available to be unlocked')}</div></div>
+                <div>
+                  {formatBalance(balancesAll.vestedClaimable, { forceUnit: '-' })}
+                  <div className='faded'>{t('available to be unlocked')}</div>
+                  {formatBalance(balancesAll.vestingPerBlock)}
+                  <div className='faded'>{t('per block')}</div>
+                  <div className='faded'>
+                    <div>{`${t('until block')} ${formatNumber(balancesAll.vestingEndBlock)}`}<div>
```
Comment (jacogr) [inline packages/react-components/src/AddressInfo.tsx:278]:
```suggestion
                    <div>{`${t('until block')} ${formatNumber(balancesAll.vestingEndBlock)}`}</div>
```

Grrr.... GH editor.

**Pull request merged**